prefect_census/syncs.py

```diff
@@ -207,3 +219,175 @@ def my_flow():
+
+        run_data = response.json()["data"]
+
+        if "sync_run_id" in run_data:
+            self.logger.info(
+                f"Census sync with ID: {self.sync_id} successfully triggered. "
+                "You can view the status of this sync run at "
+                f"https://app.getcensus.com/sync/{self.sync_id}/sync-history"
+            )
+
+        run_id = run_data["sync_run_id"]
+        if run_id is None:
+            raise RuntimeError("Unable to determine run ID for triggered sync.")
+
+        return CensusSyncRun(census_sync=self, run_id=run_id)
+
+
+class CensusSyncRun(JobRun):
+    """
+    A Job Run representing a Census sync run.
+
+    Attributes:
+        sync: The CensusSync instance that triggered this run.
+        run_id: The ID of the Census sync run.
+        run_data: The data returned by the Census API for this run.
+        status: The status of the Census sync run.
+    """
+
+    def __init__(self, census_sync, run_id) -> None:
+        self.sync = census_sync
+        self.run_id = run_id
+        self.run_data = None
+        self.status = None
+
+    @sync_compatible
+    async def wait_for_completion(self):
+        """Wait for the Census sync run to complete."""
+        seconds_waited_for_run_completion = 0
+        while seconds_waited_for_run_completion <= self.sync.max_wait_seconds:
+            try:
+                async with self.sync.credentials.get_client() as client:
+                    response = await client.get_run_info(self.run_id)
+            except HTTPStatusError as e:
+                raise RuntimeError(extract_user_message(e)) from e
+
+            run_data = response.json()["data"]
+            self.status = run_data.get("status")
+
+            if CensusSyncRunStatus.is_terminal_status_code(self.status):
+                self.run_data = run_data
+                if self.status == CensusSyncRunStatus.COMPLETED.value:
+                    self.logger.info(
+                        "Census sync run with ID %s completed successfully!",
+                        self.run_id,
+                    )
+                    return
+
+                elif self.status == CensusSyncRunStatus.CANCELLED.value:
+                    raise CensusSyncRunCancelled(
+                        f"Triggered sync run with ID {self.run_id} was cancelled."
+                    )
+                elif self.status == CensusSyncRunStatus.FAILED.value:
+                    raise CensusSyncRunFailed(
+                        f"Triggered sync run with ID: {self.run_id} failed."
+                    )
+                else:
+                    raise RuntimeError(
+                        f"Sync run with ID: {self.run_id} ended with unexpected "
+                        f"status {self.status}"
+                    )
+
+            self.logger.info(
+                "Census sync run with ID %i has status %s.",
+                self.run_id,
+                CensusSyncRunStatus(self.status).name,
+            )
```

Comment:
Depending on the length of the job, this might spam logs; I recommend storing the previous status and checking if there are changes, like:
https://github.com/PrefectHQ/prefect-dbt/pull/101/files#diff-826f2d10c2bbb1fb21d3f9003e437a53594ab7b22f01e4a75b8a14ab7c9b4fedR712-R715